Fix the following error since GCC 15:

__memory/shared_ptr.h:227:6: error: inlining failed in call to 'always_inline' 'void std::__ndk1::__shared_ptr_pointer<_Tp, _Dp, _Alloc>::__on_zero_shared() noexcept [with _Tp = Expression*; _Dp = std::__ndk1::shared_ptr<Expression>::__shared_ptr_default_delete<Expression, Expression>; _Alloc = std::__ndk1::allocator<Expression>]': --param max-inline-insns-auto limit reached
  227 | void __shared_ptr_pointer<_Tp, _Dp, _Alloc>::__on_zero_shared() _NOEXCEPT {
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/builder/.termux-build/_cache/android-r28c-api-24-aarch64-linux-android-gcc-15.1.0-r0-v0-sysroot-r0/include/c++/15.1.0/__memory/shared_ptr.h:157:23: note: called from here
  157 |       __on_zero_shared();
      |       ~~~~~~~~~~~~~~~~^~

--- a/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/__memory/shared_ptr.h
+++ b/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/__memory/shared_ptr.h
@@ -147,7 +147,8 @@
 public:
   _LIBCPP_HIDE_FROM_ABI explicit __shared_count(long __refs = 0) _NOEXCEPT : __shared_owners_(__refs) {}
 
-#if defined(_LIBCPP_SHARED_PTR_DEFINE_LEGACY_INLINE_FUNCTIONS)
+#if defined(_LIBCPP_SHARED_PTR_DEFINE_LEGACY_INLINE_FUNCTIONS) || \
+  defined(_LIBCPP_COMPILER_GCC) && defined(_LIBCPP_DEPRECATED_ABI_LEGACY_LIBRARY_DEFINITIONS_FOR_INLINE_FUNCTIONS)
   void __add_shared() noexcept;
   bool __release_shared() noexcept;
 #else
@@ -175,7 +176,8 @@
   ~__shared_weak_count() override;
 
 public:
-#if defined(_LIBCPP_SHARED_PTR_DEFINE_LEGACY_INLINE_FUNCTIONS)
+#if defined(_LIBCPP_SHARED_PTR_DEFINE_LEGACY_INLINE_FUNCTIONS) || \
+  defined(_LIBCPP_COMPILER_GCC) && defined(_LIBCPP_DEPRECATED_ABI_LEGACY_LIBRARY_DEFINITIONS_FOR_INLINE_FUNCTIONS)
   void __add_shared() noexcept;
   void __add_weak() noexcept;
   void __release_shared() noexcept;
